src/utils.js

```diff
@@ -35,15 +38,273 @@ playFeedbackAudio(true, 'correct.mp3', 'incorrect.mp3')
 playFeedbackAudio(null, 'neutral.mp3',)
 */
 
-
 export function playFeedbackAudio(responseIsCorrect, audio1, audio2) {
-    let audioToPlay
+  let audioToPlay
+
+  if (responseIsCorrect || responseIsCorrect === null) {
+    audioToPlay = audio1
+  } else {
+    audioToPlay = audio2
+  }
+
+  new Audio(audioToPlay).play()
+}
+
+// Gets the app language based on the lng query string. Using this because ROAR apps use i18next.
+function getLanguage(setLanguage) {
+  if (setLanguage) return setLanguage
+
+  let results = new RegExp('[\?&]lng=([^&#]*)').exec(window.location.href);
+  if (results == null) {
+    // default to English
+     return 'en';
+  }
+  return decodeURI(results[1]) || 0;
+}
+
+// Gets the device type based on the device inputs
+function getDevice() {
+  if (deviceType === 'touchOnly' || (deviceType === 'hybrid' && primaryInput === 'touch')) {
+    return 'mobile'
+  }
+  return 'desktop'
+}
+
+function getFormattedURL(bucketURI, filePath, lng, device, type, nested, isDefault) {
```

Comment:
Maybe you could add a jsDoc string to this function to elaborate on what `type` is as well as the other params.